### PR TITLE
refactored data download

### PR DESCRIPTION
**Pull request by xrotwang**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -105,8 +105,12 @@ ENV/
 
 # Raw data
 /raw/data/
+/raw/*_wd.csv
+/raw/*_ph.csv
+/raw/*_gloss-abbreviations.csv
+/raw/*_metadata.csv
```
Comment (xrotwang) [inline .gitignore:111]: I download individual files per corpus.

File: .gitignore [hunk 2]
```diff
@@ -105,8 +105,12 @@ ENV/
...
 /raw/wd_data.csv
 /raw/ph_data.csv
 /raw/glosses.csv
 /raw/file_metadata.csv
-/cldf/
```
Comment (xrotwang) [inline .gitignore:112]: We may have to zip the large data files to make them suitable for github.

File: cldfbench_doreco.py [hunk 3]
```diff
@@ -15,21 +20,46 @@ def cldf_specs(self):  # A dataset must declare all CLDF sets it creates.
                 module='StructureDataset'
                 )
 
+    def cmd_download(self, args):
+        self.raw_dir.download(
+            'https://sharedocs.huma-num.fr/wl/?id=s947TcfRfDZR643QURdyncdku4EmeKyb&fmode=download',
+            'languages.csv')
+        self.raw_dir.download(
+            'https://sharedocs.huma-num.fr/wl/?id=xqkUR3WoFOKB8cRb0MdHYJxIDEXEXlVG&fmode=download',
+            'sources.bib')
```
Comment (xrotwang) [inline cldfbench_doreco.py:29]: These are the two collection-level metadata files.

File: cldfbench_doreco.py [hunk 4]
```diff
@@ -15,21 +20,46 @@ def cldf_specs(self):  # A dataset must declare all CLDF sets it creates.
...
+
+        for row in self.raw_dir.read_csv('languages.csv', dicts=True):
+            if 'ND' not in row['Annotation license']:
+                print(row['Glottocode'], row['DOI'])
+                cid = urllib.parse.quote(row['DOI'], safe='')
+                for f in requests.get("{}datas/{}?metadata-format=dc".format(NAKALA_API, cid)).json()['files']:
```
Comment (xrotwang) [inline cldfbench_doreco.py:35]: Retrieve metadata about files of a corpus from nakala, using the [API](https://api.nakala.fr/doc).

File: cldfbench_doreco.py [hunk 5]
```diff
@@ -15,21 +20,46 @@ def cldf_specs(self):  # A dataset must declare all CLDF sets it creates.
...
+                    for s in ['_wd.csv', '_ph.csv', '_metadata.csv', '_gloss-abbreviations.csv']:
+                        if f['name'].endswith(s):
+                            print(f['name'], f['sha1'])
+                            self.raw_dir.download('{}data/{}/{}'.format(NAKALA_API, cid, f['sha1']), '{}{}'.format(row['Glottocode'], s))
+                            break
+
+    def iter_rows(self, pattern):
```
Comment (xrotwang) [inline cldfbench_doreco.py:42]: Here, we "merge" the files of a certain type across corpora.

File: cldfbench_doreco.py [hunk 6]
```diff
@@ -15,21 +20,46 @@ def cldf_specs(self):  # A dataset must declare all CLDF sets it creates.
...
         args.log.info("added sources")
 
-        for row in self.etc_dir.read_csv(
-            'doreco_languages_metadata.csv',
-            dicts=True,
-            ):
+        for row in self.raw_dir.read_csv('languages.csv', dicts=True):
             args.writer.objects["LanguageTable"].append({
-                "ID": row["Language"],
+                "ID": row["Glottocode"],
```
Comment (xrotwang) [inline cldfbench_doreco.py:63]: Glottocodes make for "better" identifiers in the sense that they contain only tightly controlled characters, suitable in all sorts of contexts (HTL, URL parameters, etc.)

Comment (FredericBlum) [reply cldfbench_doreco.py:63]: There is also a row `mc-zero` that is missing to some datasets. Not quite sure what it refers to, but it should probably be added as well, right? Similar to `doreco-mb-algn`

File: .gitignore (hunk 2 above)
Comment (FredericBlum) [reply .gitignore:112]: Maybe your list-based approach shows to be sufficient. We'll see.

File: cldfbench_doreco.py (hunk 5 above)
Comment (FredericBlum) [reply cldfbench_doreco.py:42]: Yup, that's way more efficient than what I had done.

File: cldfbench_doreco.py (hunk 6 above)
Comment (Tarotis) [reply cldfbench_doreco.py:62]: This refers to those lines: https://github.com/cldf-datasets/doreco/blob/robert/cldfbench_doreco.py#L46-L48

**Review by FredericBlum — APPROVED** [review]
Thanks for the rework! This makes the whole workflow much more efficient.

**Comment by FredericBlum** [comment]
I added support for the mc-zero column that is also missing to some files. I propose we merge this now, and check which aspects remain to be done.

This PR will close #6 (no more preprocessing), #8 (ID's are unique) and #5 (info about extended is part of metadata.csv).

**Pull request merged**